rxbilling/src/main/java/com/gen/rxbilling/client/RxBilling.kt

```diff
@@ -99,81 +98,104 @@ class RxBillingImpl(billingFactory: BillingClientFactory)
                 }
                 .firstOrError()
                 .flatMapCompletable {
-                    return@flatMapCompletable if (isSuccess(it)) {
+                    return@flatMapCompletable if (isSuccess(it.responseCode)) {
                         Completable.complete()
                     } else {
-                        Completable.error(BillingException.fromCode(it))
+                        Completable.error(BillingException.fromResult(it))
                     }
                 }
     }
 
-    override fun consumeProduct(purchaseToken: String): Completable {
+    override fun consumeProduct(params: ConsumeParams): Completable {
         return connectionFlowable
                 .flatMap { client ->
                     Flowable.create<Int>({
-                        client.consumeAsync(purchaseToken) { responseCode, _ ->
+                        client.consumeAsync(params) { result, _ ->
                             if (it.isCancelled) return@consumeAsync
+                            val responseCode = result.responseCode
                             if (isSuccess(responseCode)) {
                                 it.onNext(responseCode)
```

Comment:
But here i really can't realize the point of having Flowable and this `onNext` event in which the response code is emitted (which would be ignored in any case).

Reply:
I'll try to explain:
We can't use 
```
connectionFlowable
                .flatMapCompletable {  }
```

because connectionFlowable never completes in a normal flow, so we'll never get response.
Also we can't write in this way:
```
connectionFlowable
                .firstOrError()
                .flatMapCompletable {  }
```
because connectionFlowable will be canceled and billingClient.endConnection() will be called, so we'll not be able to call client.consume() anymore.

Actually, you are right, response code is not used here, but it exists just to convert flowable to single event and then to Completable. It can be replaced with Boolean or Unit, but it will look not much better I think.

Anyway, I would be grateful for a better solution.